arangod/Aql/Optimizer/Rule/OptimizerRulePushDownLateMaterialization.cpp

```diff
@@ -192,6 +192,9 @@ bool checkIndexNode(ExecutionPlan* plan,
                     ExecutionNode* other) {
   auto const* indexNode = ExecutionNode::castTo<IndexNode*>(other);
 
+  if (indexNode->options().pushDownMaterialization) {
+    return checkPossibleProjection(plan, matNode, other);
+  }
```

Comment:
I suggest creating a new function called something like `eligableForPushDownMateriaization` which returns a bool, if we want to move materialization past this for loop. (so combine it with the condition below).

Otherwise we end up with multiple branches that have to stay in sync.

Reply:
Makes sense, will do.